src/app/components/works/work4/work4.component.ts

```diff
@@ -1,15 +1,203 @@
+import { Converter } from '@core/converter';
+import { Gamma } from '@core/gamma';
+import { Avalanche } from '@core/avalanche';
+
 import { Component, OnInit } from '@angular/core';
+import { Label } from 'ng2-charts';
+import { DES } from '@/app/core/des';
+import { EDE } from '@/app/core/ede';
+
+type Convertable = 'base' | 'key1' | 'key2' | 'IVp' | 'IVc' | 'cipher';
 
 @Component({
   selector: 'app-work4',
   templateUrl: './work4.component.html',
   styleUrls: ['./work4.component.css']
 })
 export class Work4Component implements OnInit {
+  checkedEde = true;
+
+  viewModes = ['Str', 'Bin', 'Hex'];
+  views = {base: 0, cipher: 0, key1: 1, key2: 1, IVp: 1, IVc: 1};
+
+  base = 'Hello Darkness my old friend. I\'ve come to talk with you again';
+  key1 = '';
+  key2 = '';
+  IVp = '';
+  IVc = '';
+  cipher = '';
+
+  target: Convertable = 'base';
+  targetBit = '1';
+
+  lineChartData: any[] = [
+    {data: [0], label: 'Количество изменённых бит'},
+  ];
+  lineChartLabels: Label[] = ['0'];
+
+  ngOnInit() {
+    this.setLabels(16);
+  }
+
+  changeView(key: Convertable) {
+    this.views[key] = (this.views[key] + 1) % this.viewModes.length;
+    const from = this[key];
+    const to = [
+      Converter.hexToStr.bind(Converter),
+      Converter.strToBin.bind(Converter),
+      Converter.binToHex.bind(Converter)
+    ][this.views[key]](from);
+    this[key] = to;
+  }
+
+  getGamma(method: 'random' | 'scrambler' = 'random'): string {
+    return {
+      random: Gamma.getSimpleGamma(64),
+      scrambler: Gamma.getScramblerGamma(64).gamma
+    }[method];
+  }
 
-  constructor() { }
+  applyCipher(direction: 'direct' | 'reverse' = 'direct'): void {
+    const binBase = this.getBin('base');
+    const binKey1 = this.getBin('key1');
+    const binKey2 = this.getBin('key2');
+    const binIVp = this.getBin('IVp');
+    const binIVc = this.getBin('IVc');
+    let binCipher = '';
 
-  ngOnInit(): void {
+    if (this.checkedEde) {
+      const ede = new EDE(binKey1, binKey2);
+      binCipher = ede.apply(binBase, direction === 'direct');
+    } else {
+      const des = new DES(binKey1);
+      binCipher = des.pcbc(binBase, binIVc, binIVp, direction === 'direct');
+    }
+
+    this.cipher = this.fromBin('cipher', binCipher);
```

Comment:
Probably should replace direct changing to returning a value. Could be useful for avalanche plots.